app/controllers/bike_share_app.rb

```diff
@@ -1,4 +1,156 @@
+require_relative 'pagination'
+
 class BikeShareApp < Sinatra::Base
 
+  include Pagination
+
+  get '/' do
+    redirect '/stations'
+  end
+
+  get '/stations' do
+    @stations = Station.all
+    erb :"stations/index"
+  end
+
+  get '/stations-dashboard' do
+    erb :"stations/dashboard"
+  end
+
+  get '/stations/new' do
+    @cities = City.all
+    erb :"stations/new"
+  end
+
+  get '/stations/:id' do |id|
+    @station = Station.find(id)
+    erb :"stations/show"
+  end
+
+  post '/stations' do
+    city_name = params["city"]
+    city      = City.find_or_create_by(name: city_name)
+    station   = city.stations.create(params["station"])
+    redirect "/stations/#{station.id}"
+  end
+
+  get '/stations/:id/edit' do |id|
+    @station = Station.find(id)
+    @cities = City.all
+    erb :"stations/edit"
+  end
+
+  put '/stations/:id' do |id|
+    @station = Station.find(id)
+    city_name = params["city"]
+    city      = City.find_or_create_by(name: city_name)
+    @station.update(params["station"])
+    @station.update(city_id: city.id)
```

Comment:
how can we do this and the line before all in one line so we don't have to update twice?